docs/docs/concepts/auth.md

```diff
@@ -0,0 +1,291 @@
+
+1. Add metadata to be saved during resource creation by mutating the `value["metadata"]` dictionary directly.
+2. Filter resources by metadata during search/list or read operations by returning a filter dictionary.
+3. Raise an HTTP exception if access is denied.
+
+If you want to just implement simple user-scoped access control, you can use a single `@auth.on` handler for all resources and actions.
+
+```python
+@auth.on
+async def add_owner(ctx: Auth.types.AuthContext, value: dict):
+    """Add owner to resource metadata and filter by owner."""
+    filters = {"owner": ctx.user.identity}
+    metadata = value.setdefault("metadata", {})
+    metadata.update(filters)
+    return filters
+```
+
+### Resource-Specific Handlers
+
+You can register handlers for specific resources and actions using the `@auth.on` decorator.
+When a request is made, the most specific handler that matches that resource and action is called.
+
+```python
+# Generic / global handler catches calls that aren't handled by more specific handlers
+@auth.on
+async def reject_unhandled_requests(ctx: Auth.types.AuthContext, value: Any) -> None:
+    print(f"Request to {ctx.path} by {ctx.user.identity}")
+    return False
+
+# Thread creation
+@auth.on.threads.create
+async def on_thread_create(
+    ctx: Auth.types.AuthContext,
+    value: Auth.types.threads.create.value
+):
+    metadata = value.setdefault("metadata", {})
+    metadata["owner"] = ctx.user.identity
+    return {"owner": ctx.user.identity}
+
+# Thread retrieval
+@auth.on.threads.read
+async def on_thread_read(
+    ctx: Auth.types.AuthContext,
+    value: Auth.types.threads.read.value
+):
+    return {"owner": ctx.user.identity}
+
+# Run creation, streaming, updates, etc.
+@auth.on.threads.create_run
+async def on_run_create(
+    ctx: Auth.types.AuthContext,
+    value: Auth.types.threads.create_run.value
+):
+    # Inherit thread's access control
+    return {"owner": ctx.user.identity}
+
+# Assistant creation
+@auth.on.assistants.create
+async def on_assistant_create(
+    ctx: Auth.types.AuthContext,
+    value: Auth.types.assistants.create.value
+):
+    if "admin" not in ctx.user.get("role", []):
+        raise Auth.exceptions.HTTPException(
+            status_code=403,
+            detail="Only admins can create assistants"
+        )
+```
+
+Using the setup above, a request to create a `thread` would match the `on_thread_create` handler, since it is the most specific handler for that resource and action. A request to create a `cron`, on the other hand, would match the global handler, since no more specific handler is registered for that resource and action.
+
+### Filter Operations
+
+Authorization handlers can return a filter dictionary to filter resources during all operations (both reads and writes). The filter dictionary supports two additional operators:
```

Comment:
"dictionary supports two additional" ? what does `additional` mean here? 

Below it's listing 3 operators rather than 2